packages/css/src/badge.css

```diff
@@ -1,89 +1,88 @@
 .ds-badge {
-  /* color vars */
   --dsc-badge-background: var(--ds-color-base-default);
   --dsc-badge-color: var(--ds-color-contrast-default);
+  --dsc-badge-padding: 0 calc(var(--ds-spacing-1) + calc(var(--ds-spacing-1) / 2));
+  --dsc-badge-size: calc(var(--ds-sizing-3) + calc(var(--ds-spacing-1) / 2));
+
+  background: var(--dsc-badge-background);
+  border-radius: var(--ds-border-radius-full);
+  box-sizing: border-box;
+  color: var(--dsc-badge-color);
+  display: inline-grid;
+  font-size: var(--ds-font-size-minus-1); /* Default to small when size is not defined */
+  min-height: var(--dsc-badge-size);
+  min-width: var(--dsc-badge-size);
+  padding: var(--dsc-badge-padding);
+  place-items: center;
+  box-sizing: border-box;
+  line-height: var(--ds-line-height-sm);
+  width: fit-content;
 
   &[data-color='neutral'] {
     --dsc-badge-background: var(--ds-color-neutral-surface-default);
     --dsc-badge-color: var(--ds-color-neutral-text-default);
   }
+}
 
-  /* sizing vars */
-  --dsc-badge-padding: 0 calc(var(--ds-spacing-1) + calc(var(--ds-spacing-1) / 2));
-  --dsc-badge-size: calc(var(--ds-sizing-3) + calc(var(--ds-spacing-1) / 2));
-
+.ds-badge--placement {
   box-sizing: border-box;
   display: inline-flex;
```

Comment:
Maybe we should `display: inline` to avoid side effects of `flex` on children 🤔

Reply:
This is with your proposed change:
![image](https://github.com/user-attachments/assets/bf2c0697-a49a-4747-9e7c-ac175b7c5ebf)

This is with `inline-flex`:
![image](https://github.com/user-attachments/assets/a4c74487-f861-435a-bb43-729ba351dade)


Even though the SVG element only fills to a certain height, the placement wrapper wants to be bigger 🤷🏻